### PR TITLE
Openapi client migration

### PR DESCRIPTION
**Pull request by jcii**
code generation is your friend

migrated everything except the auth stuff, because it's not in the swagger spec yet.

### CODE REVIEW

File: src/app/configs/configs.service.ts [hunk 1]
```diff
@@ -29,11 +29,9 @@ export class ConfigsService {
       return of(this.configs);
     }
 
-    console.log("getAllConfigs: " + this.uriBase);
     // withCredentials: true is normally necessary, but configs are enabled for anonymous
-    return this.http.get(this.uriBase).pipe(
+    return this.client.apiConfigsGet().pipe(
       map((res) => {
```
Comment (esteban-gs) [inline src/app/configs/configs.service.ts:34]: Nit pick:
You could make it cleaner with:
```ts
.pluck("data")
.map(data => data as Config[])
.tap(data => this.configsAge = Date.now())
```

Comment (jcii) [reply src/app/configs/configs.service.ts:34]: what is this pluck and tap voodoo?

File: src/index.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <!doctype html>
-<html>
+<html lang>
```
Comment (esteban-gs) [inline src/index.html:2]: Is this supposed to be `lang="en"`?

File: src/app/configs/configs.service.ts (hunk 1 above)
Comment (esteban-gs) [reply src/app/configs/configs.service.ts:34]: [pluck is used for grabbing a value](https://www.learnrxjs.io/learn-rxjs/operators/transformation/pluck)
[tap for side effects](https://www.learnrxjs.io/learn-rxjs/operators/utility/do)

File: src/index.html (hunk 2 above)
Comment (jcii) [reply src/index.html:2]: i hacked at it to get it to not error out. I'll add lang="en"

File: src/app/configs/configs.service.ts (hunk 1 above)
Comment (jcii) [reply src/app/configs/configs.service.ts:34]: it didnt work; error-checker was fine with pluck, complained about map

File: src/app/online-orders/work-order/work-order.service.ts [hunk 3]
```diff
@@ -41,18 +42,18 @@ export class WorkOrderService {
     if (data) {
       const order: WorkOrder = JSON.parse(data) as WorkOrder;
       //console.log('get: returning stored order', order);
-      order.dateTimeofWork = new Date(order.dateTimeofWork);
+      //order.dateTimeofWork = new Date(order.dateTimeofWork);
```
[the remark this answers is not in the document]
Comment (jcii) [reply src/app/online-orders/work-order/work-order.service.ts:45]: done

File: src/index.html (hunk 2 above)
Comment (jcii) [reply src/index.html:2]: fixed

File: src/app/configs/configs.service.ts (hunk 1 above)
Comment (esteban-gs) [reply src/app/configs/configs.service.ts:34]: Sorry, it should be:

```ts
      pluck("data"),
      map((data) => data as Config[]),
      tap(() => (this.configsAge = Date.now()))
```

**Pull request merged**